src/libraries/System.Numerics.Tensors/ref/System.Numerics.Tensors.netcore.cs

```diff
@@ -4,8 +4,267 @@
+        public System.Buffers.NIndex End { get { throw null; } }
+        public System.Buffers.NIndex Start { get { throw null; } }
+        public static System.Buffers.NRange EndAt(System.Buffers.NIndex end) { throw null; }
+        public bool Equals(System.Buffers.NRange other) { throw null; }
+        public override bool Equals([System.Diagnostics.CodeAnalysis.NotNullWhenAttribute(true)] object? value) { throw null; }
+        public override int GetHashCode() { throw null; }
+        public (nint Offset, nint Length) GetOffsetAndLength(nint length) { throw null; }
+        public static explicit operator checked System.Range (System.Buffers.NRange value) { throw null; }
+        public static explicit operator System.Range (System.Buffers.NRange value) { throw null; }
+        public static implicit operator System.Buffers.NRange (System.Range range) { throw null; }
+        public static System.Buffers.NRange StartAt(System.Buffers.NIndex start) { throw null; }
+        public System.Range ToRange() { throw null; }
+        public System.Range ToRangeUnchecked() { throw null; }
+        public override string ToString() { throw null; }
+    }
+}
 namespace System.Numerics.Tensors
 {
+    public partial interface IReadOnlyTensor<TSelf, T> : System.Collections.Generic.IEnumerable<T>, System.Collections.IEnumerable where TSelf : System.Numerics.Tensors.IReadOnlyTensor<TSelf, T>
+    {
+        static abstract TSelf? Empty { get; }
+        nint FlattenedLength { get; }
+        bool IsEmpty { get; }
+        bool IsPinned { get; }
+        T this[scoped System.ReadOnlySpan<System.Buffers.NIndex> indexes] { get; }
+        TSelf this[scoped System.ReadOnlySpan<System.Buffers.NRange> ranges] { get; }
+        T this[scoped System.ReadOnlySpan<nint> indexes] { get; }
+        int Rank { get; }
+        System.Numerics.Tensors.ReadOnlyTensorSpan<T> AsReadOnlyTensorSpan();
+        System.Numerics.Tensors.ReadOnlyTensorSpan<T> AsReadOnlyTensorSpan(scoped System.ReadOnlySpan<System.Buffers.NIndex> startIndex);
+        System.Numerics.Tensors.ReadOnlyTensorSpan<T> AsReadOnlyTensorSpan(scoped System.ReadOnlySpan<System.Buffers.NRange> range);
+        System.Numerics.Tensors.ReadOnlyTensorSpan<T> AsReadOnlyTensorSpan(scoped System.ReadOnlySpan<nint> start);
+        void CopyTo(System.Numerics.Tensors.TensorSpan<T> destination);
+        void FlattenTo(System.Span<T> destination);
+        void GetLengths(System.Span<nint> destination);
+        ref readonly T GetPinnableReference();
+        void GetStrides(System.Span<nint> destination);
+        TSelf Slice(scoped System.ReadOnlySpan<System.Buffers.NIndex> startIndex);
+        TSelf Slice(scoped System.ReadOnlySpan<System.Buffers.NRange> range);
+        TSelf Slice(scoped System.ReadOnlySpan<nint> start);
+        bool TryCopyTo(System.Numerics.Tensors.TensorSpan<T> destination);
+        bool TryFlattenTo(System.Span<T> destination);
```

Comment:
Also missing `scoped` on some of these inputs. No implementation should ever capture them.